### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Pnom**

### CODE REVIEW

File: src/obis-observer/commands/uplink/GetMeterArchiveProfileResponse.ts [hunk 1]
```diff
@@ -69,22 +84,29 @@ class GetArchiveProfileResponse extends Command {
...
 
     // returns full message - header with body
     toBytes (): Uint8Array {
-        const buffer = new CommandBinaryBuffer(COMMAND_SIZE);
-        const {requestId, archiveProfile1Period, archiveProfile2Period} = this.parameters;
+        if ( !isValidParameterSet(this.parameters) ) {
+            return Command.toBytes(COMMAND_ID, new Uint8Array([this.parameters.requestId]));
+        }
+
+        const {requestId, archive1Period, archive2Period} = this.parameters as IGetArchiveProfileResponseParameters;
```
Comment (DarkPark) [inline src/obis-observer/commands/uplink/GetMeterArchiveProfileResponse.ts:105]:
```suggestion
```

File: src/obis-observer/commands/uplink/GetMeterArchiveProfileResponse.ts [hunk 2]
```diff
@@ -69,22 +84,29 @@ class GetArchiveProfileResponse extends Command {
...
 
+        const buffer = new CommandBinaryBuffer(this.size as number);
```
Comment (DarkPark) [inline src/obis-observer/commands/uplink/GetMeterArchiveProfileResponse.ts:106]:
```suggestion
        const buffer = new CommandBinaryBuffer(this.size as number);

```

File: src/obis-observer/commands/uplink/GetMeterDateResponse.ts [hunk 3]
```diff
@@ -70,26 +74,22 @@ class GetMeterDateResponse extends Command {
...
-        if ( typeof this.size !== 'number' ) {
-            throw new Error('unknown or invalid size');
-        }
-
-        const buffer = new CommandBinaryBuffer(this.size);
-        const {requestId, meterId, time2000} = this.parameters;
+        const buffer = new CommandBinaryBuffer(this.size as number);
+        const {requestId, time2000} = this.parameters;
 
         buffer.setUint8(requestId);
```
Comment (DarkPark) [inline src/obis-observer/commands/uplink/GetMeterDateResponse.ts:89]:
```suggestion
        buffer.setUint8(requestId);

```

File: src/obis-observer/commands/uplink/GetMeterReadoutStateResponse.ts [hunk 4]
```diff
@@ -33,8 +33,8 @@ interface IGetMeterReadoutStateResponseParameters extends ICommandParameters {
     overrunErrors: number
 }
```
Comment (DarkPark) [inline src/obis-observer/commands/uplink/GetMeterReadoutStateResponse.ts:34]:
```suggestion
}

```

File: src/obis-observer/commands/uplink/GetObisContentByIdResponse.ts [hunk 5]
```diff
@@ -66,24 +73,22 @@ class GetObisContentByIdResponse extends Command {
...
-            throw new Error('unknown or invalid size');
-        }
-
-        const buffer = new CommandBinaryBuffer(this.size);
+        const buffer = new CommandBinaryBuffer(this.size as number);
         const {requestId, content} = this.parameters;
 
         buffer.setUint8(requestId);
-        buffer.setFloat32(roundNumber(content));
+        if ( content ) {
```
Comment (DarkPark) [inline src/obis-observer/commands/uplink/GetObisContentByIdResponse.ts:89]:
```suggestion
        if ( content ) {

```

File: src/obis-observer/commands/uplink/GetObisProfileResponse.ts [hunk 6]
```diff
@@ -86,24 +93,23 @@ class GetObisProfileResponse extends Command {
...
-        }
-
-        const buffer = new CommandBinaryBuffer(this.size);
+        const buffer = new CommandBinaryBuffer(this.size as number);
         const {requestId, obisProfile} = this.parameters;
 
         buffer.setUint8(requestId);
-        buffer.setObisProfile(obisProfile);
+
+        if (obisProfile) {
```
Comment (DarkPark) [inline src/obis-observer/commands/uplink/GetObisProfileResponse.ts:110]:
```suggestion
        if ( obisProfile ) {
```

File: src/obis-observer/commands/uplink/GetMeterArchiveProfileResponse.ts (hunk 2 above)
Comment (Pnom) [reply src/obis-observer/commands/uplink/GetMeterArchiveProfileResponse.ts:105]: did it

File: src/obis-observer/commands/uplink/GetMeterReadoutStateResponse.ts (hunk 4 above)
Comment (Pnom) [reply src/obis-observer/commands/uplink/GetMeterReadoutStateResponse.ts]: did it

File: src/obis-observer/commands/uplink/GetObisContentByIdResponse.ts (hunk 5 above)
Comment (Pnom) [reply src/obis-observer/commands/uplink/GetObisContentByIdResponse.ts:90]: did it

File: src/obis-observer/commands/uplink/GetObisInfoResponse.ts [hunk 7]
```diff
@@ -106,43 +125,32 @@ class GetObisInfoResponse extends Command {
...
-        let size = data.length - REQUEST_ID_SIZE;
         const requestId = buffer.getUint8();
-        let obis;
-        let obisProfile;
 
-        // obis code assigned
-        if ( size > 0 ) {
-            obis = buffer.getObis();
-            size -= CommandBinaryBuffer.getObisSize(obis);
+        if (buffer.isEmpty) {
```
[the remark this answers is not in the document]
Comment (Pnom) [reply src/obis-observer/commands/uplink/GetObisInfoResponse.ts]: did it

File: src/obis-observer/commands/uplink/GetObisProfileResponse.ts (hunk 6 above)
Comment (Pnom) [reply src/obis-observer/commands/uplink/GetObisProfileResponse.ts]: did it

**Pull request merged**